client/rest-client/src/main/java/org/infinispan/client/rest/impl/okhttp/RestClientOkHttp.java

```diff
@@ -116,6 +116,8 @@ public RestClientOkHttp(RestClientConfiguration configuration) {
             } else {
                builder.protocols(Arrays.asList(Protocol.HTTP_2, Protocol.HTTP_1_1));
             }
+            // OkHttp might retry infinitely on HTTP/2.
+            builder.retryOnConnectionFailure(false);
```

Comment:
When we have HTTP/2 and close the connection on the server, OkHttp will wrap the exception into a `StreamResetException` and retry infinitely.

Reply:
Is this needed if the CANCEL frame isn't sent?

Reply:
For both cases :/ If the server sends the CANCEL frame, OkHttp creates the `StreamResetException`. If the server closes the connection, OkHttp swallows the EofException and turns it into a `StreamResetException`. So, in either case, it ends up retrying infinitely.